### PR TITLE
Feat/pledges

### LINKED ISSUE #1623: Add support for Fundraising campaigns
**Is your feature request related to a problem? Please describe.**
Adding support for fundraising campaign in talawa-admin.

**Describe the solution you'd like**
1. screens for the CRUD operation of the organization funds
2. screens for the CRUD operation of the  fund campaings.
3. screens for the CRUD operation of the fund campaigns pledges.

**Additional context**
This is campaign issue of the api-issue https://github.com/PalisadoesFoundation/talawa-api/issues/1663

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by AVtheking**
**What kind of change does this PR introduce?**
This PR add support for Campaign pledges

**Issue Number:**

Fixes #1623 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/132201033/387d37b7-ae23-46e0-9bb3-db17c924c616

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@AVtheking  Please fix the failing code cov tests.

**Comment by AVtheking** [comment]
@Cioppolo14  fixed the codecov error please assign reviewers

**Comment by AVtheking** [comment]
@Cioppolo14  @palisadoes  please assign reviews so that it can be reviewed.

**Comment by Cioppolo14** [comment]
@anwersayeed @tasneemkoushar Can you review this PR?

**Comment by palisadoes** [comment]
This is an update on the PR merging freeze:

1. In the next few hours we will be merging the `develop-userTypeFix` branch into the `develop` branch. 
1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. This merge will cause some conflicts in your PR.

We decided to do this at the beginning of the weekend to give us all time to adjust PR code and create bug fixes that may arise.

Update your code at or after midnight GMT on the morning of March 23, 2024. (5:30am IST). 

If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by palisadoes** [comment]
## The PR merging freeze is lifted.

1. We are working on bug fixes that may arise. These should be few and are being tracked here:
    1. https://github.com/orgs/PalisadoesFoundation/projects/24
1. Please update your PRs and local repos. Fix any new merge conflicts that may have occurred.
    
Background:

1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

File: src/App.tsx [hunk 1]
```diff
@@ -35,6 +35,8 @@ import { useQuery } from '@apollo/client';
 import { CHECK_AUTH } from 'GraphQl/Queries/Queries';
 import Advertisements from 'components/Advertisements/Advertisements';
 import SecuredRouteForUser from 'components/UserPortal/SecuredRouteForUser/SecuredRouteForUser';
+import FundCampaignPledge from 'screens/FundCampaignPledge/FundCampaignPledge';
+import React from 'react';
```
Comment (palisadoes) [inline src/App.tsx:39]: Fix this duplicate identifier error. Add the linting test to the eslint file:
- https://github.com/PalisadoesFoundation/talawa-admin/blob/develop/.eslintrc.json

**Review by palisadoes — CHANGES_REQUESTED** [review]
Minor fix. There is a re-import, but we need to add it to linting too.

**Comment by palisadoes** [comment]
Please fix the failing test

**Comment by palisadoes** [comment]
Please fix the formatting

**Comment by AVtheking** [comment]
@palisadoes  all tests are passing now

**Comment by Cioppolo14** [comment]
@AVtheking The test application check is not passing.

**Comment by AVtheking** [comment]
@Cioppolo14  @palisadoes now all tests are passing.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**